hermes-common/src/main/java/pl/allegro/tech/hermes/common/hook/HooksHandler.java

```diff
@@ -21,24 +23,35 @@ public void addShutdownHook(ServiceAwareHook hook) {
     }
 
     public void shutdown(ServiceLocator serviceLocator) {
+        deregisterGlobalShutdownHook();
```

Comment:
It is related with graceful shutdown from tests level. If you call in tests `hermes{Consumers,Frontend}.stop()` then it does not release global shutdown hook which fires again all shutdown hooks when VM is shutting down. I discovered it when I was writting jmh tests. JMH spawns new VM for every benchmark, and I saw that VM was waiting [here](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/lang/ApplicationShutdownHooks.java?av=f#106) after each finished test